components/user_profile/user_profile.tsx

```diff
@@ -1,52 +1,56 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React, {PureComponent} from 'react';
 
+import {UserProfile as UserProfileType} from 'mattermost-redux/types/users';
+
 import {imageURLForUser, isMobile, isGuest} from 'utils/utils.jsx';
 
-import OverlayTrigger from 'components/overlay_trigger';
+import OverlayTrigger, {BaseOverlayTrigger} from 'components/overlay_trigger';
 import ProfilePopover from 'components/profile_popover';
 import BotBadge from 'components/widgets/badges/bot_badge';
 import GuestBadge from 'components/widgets/badges/guest_badge';
 
-export default class UserProfile extends PureComponent {
-    static propTypes = {
-        disablePopover: PropTypes.bool,
-        displayName: PropTypes.string,
-        displayUsername: PropTypes.bool,
-        hasMention: PropTypes.bool,
-        hideStatus: PropTypes.bool,
-        isBusy: PropTypes.bool,
-        isRHS: PropTypes.bool,
-        overwriteName: PropTypes.node,
-        overwriteIcon: PropTypes.node,
-        user: PropTypes.object,
-        userId: PropTypes.string,
-    };
-
-    static defaultProps = {
+export type UserProfileProps = {
+    userId: string;
+    displayName?: string;
+    isBusy?: boolean;
+    overwriteName?: React.ReactNode;
+    overwriteIcon?: React.ReactNode;
+    user?:UserProfileType;
```

Comment:
Nit: spacing between `user?:` and `UserProfileType`

Reply:
I think ESLint might catch that now once the PR is updated